File: src/api/admin/systemVariable.ts
import { post } from '@/utils/request'

export function getList<T>(data: Common.ListRequest) {
  return post<T>({
    url: '/admin/systemVariable/getList',
    data,
  })
}

export function edit<T>(data: SystemVariable.SystemVariableEditReq) {
  return post<T>({
    url: '/admin/systemVariable/edit',
    data,
  })
}

export function set<T>(name: string, value: string) {
  return post<T>({
    url: '/admin/systemVariable/set',
    data: {
      name,
      value,
    },
  })
}

export function deleteByName<T>(name: string) {
  return post<T>({
    url: '/admin/systemVariable/delete',
    data: {
      name,
    },
  })
}

export function clearCache<T>(varName: string) {
  return post<T>({
    url: '/admin/systemVariable/clearCache',
    data: { Name: varName },
  })
}

export function getByCache<T>(varName: string) {
  return post<T>({
    url: '/admin/systemVariable/getByCache',
    data: { Name: varName },
  })
}

File: service/router/admin/systemVariable.go
package admin

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitSystemVariableRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiAdmin.SystemVariableApi

	r := router.Group("", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		r.POST("systemVariable/getList", api.GetList)
		r.POST("systemVariable/set", api.Set)
		r.POST("systemVariable/edit", api.Edit)
		r.POST("systemVariable/delete", api.Delete)
		r.POST("systemVariable/clearCache", api.ClearCache)
		r.POST("systemVariable/getByCache", api.GetByCache)
	}

}
